onnxruntime/core/providers/coreml/model/model.h

```diff
@@ -66,6 +69,10 @@ class Model {
   void SetScalarOutputs(std::unordered_set<std::string>&& scalar_outputs) {
     scalar_outputs_ = std::move(scalar_outputs);
   }
+
+  void SetInt64Outputs(std::unordered_set<std::string>&& int64_outputs) {
+    int64_outputs_ = std::move(int64_outputs);
+  }
 };
```

Comment:
Is there a next planned step (separate PR) to allow pushing the int64 boundary further along to maximize the nodes in each partition or do we not have any models where that would help?

i.e. if the node after the ArgMax is supported by CoreML using int32 (vs. int64 for the onnx node), and is valid to run using the int32 data without casting to int64, add it to the partition, and add its outputs to the Int64 outputs. e.g. ArgMax -> Gather

Reply:
Yes. Thanks for the suggestion. I will think about to do some follow up to push the handling further.

Reply:
Lower priority until we have a use case that requires this ability though.